File: chat-uikit/src/main/java/io/agora/chat/uikit/conversation/interfaces/IConversationListLayout.java
package io.agora.chat.uikit.conversation.interfaces;


import io.agora.chat.uikit.conversation.adapter.EaseConversationListAdapter;
import io.agora.chat.uikit.conversation.model.EaseConversationInfo;
import io.agora.chat.uikit.conversation.presenter.EaseConversationPresenter;
import io.agora.chat.uikit.interfaces.IRecyclerView;

public interface IConversationListLayout extends IRecyclerView {

    void setPresenter(EaseConversationPresenter presenter);

    /**
     * Whether to show the default entry menu
     * @param showDefault
     */
    void showItemDefaultMenu(boolean showDefault);

    /**
     * Set custom list adapter
     * @param listAdapter
     */
    void setListAdapter(EaseConversationListAdapter listAdapter);

    /**
     * Get data adapter
     * @return
     */
    EaseConversationListAdapter getListAdapter();

    /**
     * Get item data
     * @param position
     * @return
     */
    EaseConversationInfo getItem(int position);


    /**
     * Make conversation read
     * @param position
     * @param info
     */
    void makeConversionRead(int position, EaseConversationInfo info);

    void makeConversationTop(int position, EaseConversationInfo info);

    void cancelConversationTop(int position, EaseConversationInfo info);

    /**
     * Delete conversation
     * @param position
     * @param info
     */
    void deleteConversation(int position, EaseConversationInfo info);

    /**
     * Set up monitoring of session changes
     * @param listener
     */
    void setOnConversationChangeListener(OnConversationChangeListener listener);

    /**
     * Set up the loading session state monitor
     * @param loadListener
     */
    void setOnConversationLoadListener(OnConversationLoadListener loadListener);
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/conversation/interfaces/IConversationListLayout.kt
package io.agora.uikit.feature.conversation.interfaces

import io.agora.uikit.feature.conversation.adapter.EaseConversationListAdapter
import io.agora.uikit.common.interfaces.IRecyclerView
import io.agora.uikit.model.EaseConversation
import io.agora.uikit.viewmodel.conversations.IConversationListRequest

/**
 * Defines the conversation interface called by developers.
 */
interface IConversationListLayout: IRecyclerView {

    fun setViewModel(viewModel: IConversationListRequest?)

    /**
     * Set custom list adapter
     * @param listAdapter
     */
    fun setListAdapter(listAdapter: EaseConversationListAdapter?)

    /**
     * Get data adapter
     * @return
     */
    fun getListAdapter(): EaseConversationListAdapter?

    /**
     * Get item data
     * @param position
     * @return
     */
    fun getItem(position: Int): EaseConversation?


    /**
     * Make conversation read
     * @param position
     * @param info
     */
    fun makeConversionRead(position: Int, info: EaseConversation?)

    fun makeConversationTop(position: Int, info: EaseConversation?)

    fun cancelConversationTop(position: Int, info: EaseConversation?)

    /**
     * Delete conversation
     * @param position
     * @param info
     */
    fun deleteConversation(position: Int, info: EaseConversation?)

    /**
     * Set up monitoring of session changes
     * @param listener
     */
    fun setOnConversationChangeListener(listener: OnConversationListChangeListener?)

    /**
     * load conversation data listener
     * @param listener
     * @return
     */
    fun setLoadConversationListener(listener: OnLoadConversationListener)

}